suite-native/module-device-settings/src/hooks/useSettingsProtection.ts

```diff
@@ -0,0 +1,45 @@
+import { useCallback, useEffect } from 'react';
+import { useSelector } from 'react-redux';
+
+import { useNavigation } from '@react-navigation/native';
+
+import { selectIsDeviceConnected, selectIsDeviceDiscoveryActive } from '@suite-common/wallet-core';
+import {
+    AuthorizeDeviceStackRoutes,
+    DeviceStackRoutes,
+    RootStackParamList,
+    RootStackRoutes,
+    SettingsStackParamList,
+    StackToStackCompositeNavigationProps,
+} from '@suite-native/navigation';
+
+type NavigationProps = StackToStackCompositeNavigationProps<
+    SettingsStackParamList,
+    DeviceStackRoutes.DeviceSettings,
+    RootStackParamList
+>;
+
+export const useSettingsProtection = (isProtectionEnabled = false) => {
+    const navigation = useNavigation<NavigationProps>();
+
+    const isDiscoveryRunning = useSelector(selectIsDeviceDiscoveryActive);
+    const isDeviceConnected = useSelector(selectIsDeviceConnected);
+
+    const handleConnectDevice = useCallback(() => {
```

Comment:
I would sugest to use some better name. I would expect that function called `handleConnectDevice` will be handling the connection, but this is not the case. 

What about something like this? 
```suggestion
    const redirectToConnect&UnlockScreen = useCallback(() => {
```

Or you can remove the callback completely and move its body directly to the useEffect.

Reply:
You are right, this is a better name. I’ve just reused a function with the same responsibility. Will change